### PR TITLE
feat: revamp integration flow

### PR DESCRIPTION
**Pull request by Vikrant2520**
### Summary

- Now when the user visits the integrations page first time on click of `connect` button we open up a modal that says you need to configure... and when the user clicks `I have already configured` we make the API call to install the integration, else if he chooses to goto configure tab we do the same 
- Next when in test connection we provide two options to either wait or else go to configuration tab
- renamed the modal headings to `Connect Mongo` and `Test Mongo Connection`
- update the color of the `Test Connection` button to grey

#### Related Issues / PR's

Fixes:- https://github.com/SigNoz/engineering-pod/issues/1242

#### Screenshots

https://github.com/SigNoz/signoz/assets/54737045/af852281-0fa0-4da4-8f2b-0798cd55ace6

https://github.com/SigNoz/signoz/assets/54737045/5c2fcf6a-ff17-4454-b7a8-37237226c0bd

### CODE REVIEW

**Review by makeavish — APPROVED** [review]

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx [hunk 1]
```diff
@@ -153,15 +163,67 @@ function IntegrationDetailHeader(
...
+								if (isConnectionStateNotInstalled) {
+									setActiveDetailTab('configuration');
+								}
+								handleOk();
+							}}
+							className="understandBtn"
+						>
+							{isConnectionStatePending
+								? isConnectionStateNotInstalled
+									? 'Show Configuration Steps'
```
Comment (YounixM) [inline frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx:194]: Move to copies

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx [hunk 2]
```diff
@@ -210,12 +272,26 @@ function IntegrationDetailHeader(
 					) : connectionState === ConnectionStates.TestingConnection ? (
 						<div className="data-test-connection">
 							<div className="last-data">
-								After adding the {title} integration, you need to manually configure
-								your Redis data source to start sending data to SigNoz.
+								We have not received data from your {title} Instance yet. You need to
```
Comment (YounixM) [inline frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx:275]: Move to copies

**Review by YounixM — APPROVED** [review]

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx (hunk 1 above)
Comment (Vikrant2520) [reply frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx:194]: making the change in next PR for this

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx (hunk 2 above)
Comment (Vikrant2520) [reply frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx:275]: making the change in next PR for this

**Pull request merged**